Multisig/Data/Services/Notification Service/RegisterNotificationTokenRequest.swift

```diff
@@ -7,27 +7,58 @@
 //
 
 import Foundation
+import Web3
 
 struct RegisterNotificationTokenRequest: JSONRequest {
-    let uuid: String?
+    let uuid: String
     let safes: [String]
     let cloudMessagingToken: String
     let bundle: String
     let version: String
     let deviceType: String = "IOS"
     let buildNumber: String
+    let timestamp: String
+    let signatures: [String]?
+
     var httpMethod: String { return "POST" }
     var urlPath: String { return "/api/v1/notifications/devices/" }
 
     typealias ResponseType = Response
 
-    init(deviceID: UUID? = nil, safes: [Address], token: String, bundle: String, version: String, buildNumber: String) {
-        self.uuid = deviceID?.uuidString.lowercased()
+    init(deviceID: UUID,
+         safes: [Address],
+         token: String,
+         bundle: String,
+         version: String,
+         buildNumber: String,
+         timestamp: String = String(Int(Date().timeIntervalSince1970 * 1_000))) throws {
```

Comment:
I would move this conversion from date to string inside of the init. Can we make timestamp as Date or as TimeInterval?

Reply:
Also, this argument should be optional

Reply:
Having it as string in init param is easier for unit testing and it should generally not be passed manually in production code. Is it ok if I leave it like this?

Reply:
fine